arraylist.c

```diff
@@ -125,6 +125,27 @@ int array_list_shrink(struct array_list *arr, size_t empty_slots)
 	return 0;
 }
 
+int array_list_insert_idx(struct array_list *arr, size_t idx, void *data)
+{
+	size_t move_amount;
+
+	if (idx > arr->length)
```

Comment:
This should be "if (idx >= arr->length)"

Reply:
the code below seems to work fine to cover the `idx == arr->length`;
but it makes sense to do `idx >= arr->length` here

so, ack